src/main/webapp/task.jsp

```diff
@@ -109,23 +115,60 @@
         %>
         <form id="toggle-user-assignment-post-form" action="<%=servletPage%>" method="POST">
           <input type="hidden" name="taskID" value="<%=task.getTaskID()%>"/>
-          <input type="hidden" name="userID" value="<%=userID%>"/>
+          <input type="hidden" name="userID" value="<%=user.getUserID()%>"/>
           <button type="submit" id="toggle-user-assignment"><%=userButtonText%></button>
         </form>
         <%}%>
       <div id="task-users-container"></div>
 
       <h2>Comments</h2>
+      <div id="task-addcomments-container">
+        <%
+            if (task.getTaskID() != 0 && task.getStatus() != Status.COMPLETE) {
+        %>
+        <form id="add-comment-post-form" action="/comment" method="POST">
+          <input type="hidden" id="add-comment-task-input" name="taskID" value="<%=task.getTaskID()%>">
+          <input type="hidden" id="add-comment-user-input" name="userID" value="<%=user.getUserID()%>">
+          <input type="text" name="title" required placeholder="Enter title of comment..." maxlength="40">
+          <br>
+          <br>
+          <textarea type="text" name="message" required placeholder="Enter comment message..."></textarea>
+          <br>
+          <button type="submit">Post Comment</button>
+        </form>
+        <%}%>
+      </div>
       <ul id="task-comments-container">
+        <%
+            for (Map.Entry<CommentData, String> entry : comments.entrySet()) {
+                CommentData comment = entry.getKey();
+                String username = entry.getValue();
+                
+                // Get timestamp
+                String datePattern = "MMM d, yyyy";
+                SimpleDateFormat dateFormat = new SimpleDateFormat(datePattern);
+                String date = dateFormat.format(comment.getTimestamp());
+                String timePattern = "HH:mm";
+                SimpleDateFormat timeFormat = new SimpleDateFormat(timePattern);
+                String time = timeFormat.format(comment.getTimestamp());
+        %>
         <li class="comment">
-          <p>Comment 1</p>
-        </li>
-        <li class="comment">
-          <p>Comment 2</p>
-        </li>
-        <li class="comment">
-          <p>Comment 3</p>
+          <h3><%=comment.getTitle()%></h3>
+          <h5>
+            Posted on <%=date%> at <%=time%>
+            <br>
+            Posted by <%=username%>
+          </h5>
+          <p><%=comment.getMessage()%></p>
+          <% if (comment.getUserID() == user.getUserID() && task.getStatus() != Status.COMPLETE) {%>
+          <form id="delete-comment-post-form" action="/comment-delete" method="POST">
+            <input type="hidden" id="delete-comment-commentID-input" name="commentID" value="<%=comment.getCommentID()%>">
+            <input type="hidden" id="delete-comment-taskID-input" name="taskID" value="<%=task.getTaskID()%>">
+            <button type="submit"><span class="fa fa-trash-o" aria-hidden="true"></span></button>
```

Comment:
this could be a good use case for an ajax delete so you won't have to refresh the whole page after deleting a comment.  `remove()` I think is the function to remove from DOM?  https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove